python/ray/air/_internal/util.py

```diff
@@ -32,7 +32,9 @@ def is_nan_or_inf(value):
 class StartTraceback(Exception):
     """These exceptions (and their tracebacks) can be skipped with `skip_exceptions`"""
 
-    pass
+    def __init__(self, *args: object, tags: Optional[Dict] = None) -> None:
+        super().__init__(*args)
+        self.tags = tags if tags else {}
```

Comment:
It seems a bit weird to add these tags arbitrarily here...

IMO it might be better to have a subclass that's specific to worker failures that requires the worker_rank to be set. This way it's more explicit how these are propagated.

cc @justinvyu does this make sense?

Reply:
Do you mean subclass the `StartTraceback`?

or we can also do

```python
class StartTraceback(Exception):
  def __init__(self, *args: object, worker_rank = None) -> None:
          super().__init__(*args)
          self.worker_rank = worker_rank
```

Reply:
Yeah I mean to subclass `StartTraceback`. The reason is that the `StartTraceback` is a "common" class used generically across a few of places, so adding `worker_rank` directly to it doesn't make sense.

Also we want to make it clear that the newly added code that reads the `worker_rank` from the exception is reading specifically from the exception that has this value populated, and not another instance of `StartTraceback` that doesn't have this value populated.

Reply:
ok, i can make it a subclass... 

but this `StartTraceback` stuff will not be used in Train V2 right?